Display which config is used in --more_version

hwangcc23 opened the pull request:
Display the path of the used config and its last modified time in the output of i3 --moreversion.
See the issue #1785 for the discussion and the usage.

stapelberg commented on docs/ipc:
```diff
@@ -582,11 +582,14 @@ human_readable (string)::
 	build date and branch name. When you need to display the i3 version to
 	your users, use the human-readable version whenever possible (since
 	this is what +i3 --version+ displays, too).
+used_config (string)::
```
Can you rename this to the more descriptive `loaded_config_file_name` please?

stapelberg commented on src/display_version.c:
```diff
@@ -104,6 +109,16 @@ void display_running_version(void) {
+    char mtime[64];
+    mtime[0] = '\0';
+    if (stat(used_config, &sb) == -1) {
+        ELOG("Cannot stat config file %s\n", used_config);
```
Please use `\"%s\"` to clearly separate the path in the error message.

stapelberg commented on src/display_version.c:
```diff
@@ -104,6 +109,16 @@ void display_running_version(void) {
+    if (stat(used_config, &sb) == -1) {
+        ELOG("Cannot stat config file %s\n", used_config);
+    } else {
+        strftime(mtime, sizeof(mtime), "(last modified: %Y-%m-%d %H:%M)", localtime(&(sb.st_mtime)));
```
Instead of hardcoding a date/time format, please use `%c` which prints the time according to the current locale.

Also, can you make the code also print how many seconds ago that was? That way, when people provide the output, outsiders in different time zones can easily spot whether a stale config file was used.

stapelberg commented on src/display_version.c:
```diff
@@ -104,6 +109,16 @@ void display_running_version(void) {
+    } else {
+        strftime(mtime, sizeof(mtime), "(last modified: %Y-%m-%d %H:%M)", localtime(&(sb.st_mtime)));
+    }
+    printf("\rUsed i3 config: %s %s\n", used_config, mtime);
```
The `\r` in the beginning is not necessary, it’s only required in the earlier message because we need to clear the progress message that is printed in line 72.

hwangcc23 committed:
Display which config is used in --more_version

In the output of i3 --moreversion,
display the path of the used config and its last modified time.

hwangcc23 commented:
@stapelberg Done. Please review the patch again. Thanks.

stapelberg commented on src/display_version.c:
```diff
@@ -104,6 +109,19 @@ void display_running_version(void) {
 
     printf("\rRunning i3 version: %s (pid %s)\n", human_readable_version, pid_from_atom);
 
+    struct stat sb;
```
Please introduce an `if` statement that skips this code if loaded_config_file_name is not present in i3’s JSON reply. This is to cover the case where the i3 binary is newer than the i3 binary the user is running.

hwangcc23 commented:
@stapelberg Done. Please review it. Thanks.
